src/crimson/os/seastore/btree/btree_range_pin.h

```diff
@@ -256,8 +257,8 @@ class btree_range_pin_t : public boost::intrusive::set_base_hook<> {
 template <typename node_bound_t>
 class btree_pin_set_t {
   friend class btree_range_pin_t<node_bound_t>;
-  using pins_t = typename btree_range_pin_t<node_bound_t>::index_t;
-  pins_t pins;
+  using pins_by_depth_t = std::vector<typename btree_range_pin_t<node_bound_t>::index_t>;
+  pins_by_depth_t pins_by_depth;
```

Comment:
Since we're making a static assumption about the array size, simpler to just use a std::array with a static length and avoids an indirection.